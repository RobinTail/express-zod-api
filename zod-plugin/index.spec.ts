import { z } from "zod";
import * as entrypoint from "./index";

describe("Entrypoint", () => {
  test("Augmentation", () => {
    expectTypeOf<z.ZodAny>()
      .toHaveProperty("example")
      .toEqualTypeOf<(value: any) => z.ZodAny>();
    expectTypeOf<z.ZodDefault<z.ZodString>>()
      .toHaveProperty("example")
      .toEqualTypeOf<(value: string) => z.ZodDefault<z.ZodString>>();
    expectTypeOf<z.ZodDefault<z.ZodString>>()
      .toHaveProperty("label")
      .toEqualTypeOf<(value: string) => z.ZodDefault<z.ZodString>>();
    expectTypeOf<z.ZodObject>().toHaveProperty("remap");
  });

  test("Exports", () => {
    expect(entrypoint).toMatchObject({
      getBrand: expect.any(Function),
      pack: expect.any(Function),
      unpack: expect.any(Function),
    });
  });
});
