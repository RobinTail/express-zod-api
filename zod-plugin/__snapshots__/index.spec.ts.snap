// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Entrypoint > Exports 1`] = `
{
  "getBrand": [Function],
}
`;
